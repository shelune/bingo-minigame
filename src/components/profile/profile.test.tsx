import React from "react";
import { screen, fireEvent, render } from "@testing-library/react";

import Profile, { PlayerData } from "./profile";

const mockPlayerData: PlayerData = {
  name: "foobar",
  gamesCount: 1,
  totalTimeSpent: 9,
  recentGames: [{ winCondition: "column", turns: 13, elapsedTime: 9 }],
};

describe("Profile", () => {
  it("should render", () => {
    const { container } = render(
      <Profile
        setOpen={jest.fn()}
        isOpen
        playerData={mockPlayerData}
        deleteProfile={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("shouldn't show if not isOpen", () => {
    render(
      <Profile
        setOpen={jest.fn()}
        isOpen={false}
        playerData={mockPlayerData}
        deleteProfile={jest.fn()}
      />
    );
    expect(screen.getByTestId("profile-drawer")).not.toHaveClass("isOpen");
  });

  it("should show empty state if no playerData given", () => {
    const { container } = render(
      <Profile
        setOpen={jest.fn()}
        isOpen
        playerData={null}
        deleteProfile={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call deleteProfile if button is clicked", () => {
    const mockDeleteProfile = jest.fn();
    render(
      <Profile
        setOpen={jest.fn()}
        isOpen={false}
        playerData={mockPlayerData}
        deleteProfile={mockDeleteProfile}
      />
    );
    screen.getByText("Delete Profile").click();
    expect(mockDeleteProfile).toHaveBeenCalled();
  });
});
