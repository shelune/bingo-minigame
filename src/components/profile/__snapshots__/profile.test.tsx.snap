// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render 1`] = `
<div>
  <div
    class="content isOpen"
    data-testid="profile-drawer"
  >
    <div
      class="profileHeader"
    >
      <div
        class="closeButton"
        role="button"
      >
        &gt; Close
      </div>
      <h2>
        Profile
      </h2>
    </div>
    <div
      class="profileBody"
    >
      <h3>
        foobar
        's statistics
      </h3>
      <div>
        <div>
          Total games played: 
          1
        </div>
        <div>
          Total time played (seconds): 
          9
          s
        </div>
        <div>
          <span>
            Latest 5 games summary:
          </span>
          <ul>
            <li>
              Won by
               
              <span
                class="hightlight"
              >
                column
              </span>
               
              in 
              <span
                class="hightlight"
              >
                13
              </span>
               
              turns (
              <span
                class="hightlight"
              >
                9
              </span>
               
              seconds)
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="profileFooter"
    >
      <button
        class="button"
        type="button"
      >
        Delete Profile
      </button>
    </div>
  </div>
</div>
`;

exports[`Profile should show empty state if no playerData given 1`] = `
<div>
  <div
    class="content isOpen"
    data-testid="profile-drawer"
  >
    <div
      class="profileHeader"
    >
      <div
        class="closeButton"
        role="button"
      >
        &gt; Close
      </div>
      <h2>
        Profile
      </h2>
    </div>
    <div
      class="profileError"
    >
      No record of this player yet. Please play some games first!
    </div>
  </div>
</div>
`;
